snforge_std/src/errors.cairo

```diff
@@ -0,0 +1,31 @@
+use core::option::OptionTrait;
+use core::starknet::SyscallResult;
+use core::byte_array::BYTE_ARRAY_MAGIC;
+
+pub enum PanicDataOrString {
+    String: ByteArray,
+    PanicData: Array<felt252>,
+}
+
+type ExtendedSyscallResult<T> = Result<T, PanicDataOrString>;
+
+trait SyscallResultStringErrorTrait<T> {
+    fn map_string_errors(self: SyscallResult<T>) -> ExtendedSyscallResult<T>;
+}
+
+impl SyscallResultStringErrorTraitImpl<T> of SyscallResultStringErrorTrait<T> {
+    fn map_string_errors(self: SyscallResult<T>) -> ExtendedSyscallResult<T> {
+        match self {
+            Result::Ok(x) => Result::Ok(x),
+            Result::Err(panic_data) => {
+                if panic_data.len() > 0 && *panic_data.at(0) == BYTE_ARRAY_MAGIC {
+                    let mut panic_data_span = panic_data.span().slice(1, panic_data.len() - 1);
+                    let deserialized = Serde::<ByteArray>::deserialize(ref panic_data_span)
+                        .expect('panic string not deserializable');
```

Comment:
Shouldn't we return `ResultErr(PaniDataOrString::PanicData(..))` here then?

Reply:
It would have to be wrapped in one more Result

Reply:
If the magic byte is there and it's unparseable, it's a bug on our side